otcextensions/sdk/dms/v1/instance.py

```diff
@@ -0,0 +1,118 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+from openstack import resource
+
+from otcextensions.sdk.dms.v1 import _base
+
+
+class Instance(_base.Resource):
+    """DMS Instance resource"""
+    resources_key = 'instances'
+    base_path = '/instances'
+
+    _query_mapping = resource.QueryParameters(
+        'engine', 'name', 'status', 'include_failure',
+        'exact_match_name',
+        include_failure='includeFailure',
```

Comment:
include_failure and exact_match_name parameters support true or false values.

Reply:
I know, but we do not care about this here (actually we can't do any kind of type enforcing on this level). Those are not real attributes of the resource, therefore we can't properly check the input